src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -409,6 +433,16 @@ ColumnsDescription StorageObjectStorage::resolveSchemaFromData(
     std::string & sample_path,
     const ContextPtr & context)
 {
+    if (configuration->isDataLakeConfiguration())
+    {
+        configuration->update(object_storage, context);
```

Comment:
So we call `configuration->update` twice during storage creation in case of schema inference for data lakes. First time in `StorageObjectStorage` constructor, next time here. Maybe we can avoid it? For example update metadata inside `tryGetTableStructureFromMetadata` only if it's not initialized?